### PR TITLE
Deprecate Compat/Specifications macro parameters

### PR DESCRIPTION
**Pull request by sideshowbarker**
In current MDN article content, we no longer have any `Compat` or `Specifications` macros which have arguments — instead all articles use the `browser-compat` frontmatter key or the `spec-urls` frontmatter key.

And in no cases is it any longer necessary to ever call the `Compat` or `Specifications` macros with any arguments — instead in all cases, the `browser-compat` and `spec-urls` frontmatter keys can be used; therefore, it’s appropriate to report flaws for any new cases where a PR adds content that has arguments in `Compat` or `Specifications` macro calls.

### CODE REVIEW

**Comment by Elchi3** [comment]
Looks like test need to be updated for this (kitchensink page)

**Comment by caugner** [comment]
> Looks like test need to be updated for this (kitchensink page)

Should be fixed in d111a551cd7771250844b9ca9dfed575e079af5f.

File: kumascript/macros/Compat.ejs [hunk 1]
```diff
@@ -30,6 +30,11 @@ Example calls
 
 */
 
+if ($0) {
+  throw new Error(`Calling the Compat macro with any arguments is
+  deprecated; instead use the 'browser-compat' front-matter key.`);
```
Comment (caugner) [inline kumascript/macros/Compat.ejs:35]: Could we reuse `mdn.deprecated()` from `kumascript/src/api/mdn.js` by adding an optional `message` parameter?

```
  deprecated(message = "This macro has been deprecated, and should be removed.") {
    this.env.recordNonFatalError(
      "deprecated",
      message
    );
  },
```

File: kumascript/macros/Specifications.ejs [hunk 2]
```diff
@@ -13,6 +13,12 @@ Example calls
 
 */
 
+if ($0) {
+  throw new Error(`Calling the Specifications macro with any arguments is
+  deprecated; instead use either the 'browser-compat' or 'spec-urls'
+  front-matter key.`);
```
Comment (caugner) [inline kumascript/macros/Specifications.ejs:19]: ditto

**Review by caugner — CHANGES_REQUESTED** [review]

File: kumascript/macros/Compat.ejs (hunk 1 above)
Comment (caugner) [reply kumascript/macros/Compat.ejs:35]: @sideshowbarker If you merge the latest `main`, you will be able to pass a message to `mdn.deprecated()`.

**Review by caugner — CHANGES_REQUESTED** [review]

**Comment by sideshowbarker** [comment]
@caugner Thanks much — I just now pushed an update that switches this patch to instead calling `mdn.deprecated()` with a message parameter.

**Review by caugner — CHANGES_REQUESTED** [review]
Compat.test.js and Specifications.test.js need to be updated to pass `browser-compat` and/or `spec-urls` via the page metadata, rather than via a parameter.

**Comment by sideshowbarker** [comment]
> Compat.test.js and Specifications.test.js need to be updated to pass `browser-compat`

OK, thanks — I just now pushed an update which changes them to do that

> and/or `spec-urls` via the page metadata, rather than via a parameter

I added tests for the `spec-urls` case

**Comment by caugner** [comment]
@sideshowbarker Okay, now we only need to migrate our kitchensink page, moving the bcd key into the frontmatter:

https://github.com/mdn/content/blob/5fa776ca3c79d0298cdf07c50ed0024d8b185716/files/en-us/mdn/kitchensink/index.md?plain=1#L241

**Comment by sideshowbarker** [comment]
> @sideshowbarker Okay, now we only need to migrate our kitchensink page, moving the bcd key into the frontmatter:
> 
> [mdn/content@`5fa776c`/files/en-us/mdn/kitchensink/index.md?plain=1#L241](https://github.com/mdn/content/blob/5fa776ca3c79d0298cdf07c50ed0024d8b185716/files/en-us/mdn/kitchensink/index.md?plain=1#L241)

ah, ok, yes — just now opened https://github.com/mdn/content/pull/20480 for that

**Comment by sideshowbarker** [comment]
With https://github.com/mdn/content/pull/20480 merged, it looks like CI here is all green now ✅

**Review by caugner — APPROVED** [review]

**Pull request merged**